src/main/java/com/testmonitor/actions/TestCaseFolders.java

```diff
@@ -0,0 +1,186 @@
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class TestCaseFolders {
+    private final Connector connector;
+
+    private final Integer projectId;
+
+    /**
+     * @param connector The TestMonitor connector
+     * @param project The TestMonitor project
+     */
+    public TestCaseFolders(Connector connector, Project project)
+    {
+        this.connector = connector;
+        this.projectId = project.getId();
+    }
+
+    /**
+     * @return A list of test case folders
+     */
+    public ArrayList<TestCaseFolder> list() throws IOException, URISyntaxException {
+        return this.list(1);
+    }
+
+    /**
+     * @param page Page number
+     *
+     * @return A list of test case folders
+     */
+    public ArrayList<TestCaseFolder> list(Integer page) throws IOException, URISyntaxException {
+        return this.list(0, page, 15);
+    }
+
+    /**
+     * @param page Page number
+     * @param limit Paging limit
+     *
+     * @return A list of test case folders
+     */
+    public ArrayList<TestCaseFolder> list(Integer page, Integer limit) throws IOException, URISyntaxException {
+        return this.list(0, page, limit);
+    }
+
+    /**
+     * @param parentId test case folder parent id
+     * @param page Page number
+     * @param limit Paging limit
+     *
+     * @return A list of test case folders
+     */
+    public ArrayList<TestCaseFolder> list(Integer parentId, Integer page, Integer limit) throws IOException, URISyntaxException {
+        List<NameValuePair> params = new ArrayList<>();
+
+        params.add(new BasicNameValuePair("project_id", this.projectId.toString()));
+        params.add(new BasicNameValuePair("parent_id", parentId.toString()));
+        params.add(new BasicNameValuePair("page", page.toString()));
+        params.add(new BasicNameValuePair("limit", limit.toString()));
+
+        return TestCaseFolderParser.parse(this.connector.get("test-case/folders", params));
+    }
+
+    /**
+     * @param id The test case folder ID
+     *
+     * @return The test case folder matching the ID
+     */
+    public TestCaseFolder get(Integer id) throws IOException, URISyntaxException {
+        List<NameValuePair> params = new ArrayList<>();
+
+        params.add(new BasicNameValuePair("project_id", this.projectId.toString()));
+
+        JSONObject response = this.connector.get("test-case/folders/" + id, params);
+
+        HashMap<String, Object> testCaseFolder = (HashMap<String, Object>) response.getJSONObject("data").toMap();
+
+        return TestCaseFolderParser.parse(testCaseFolder);
+    }
+
+    /**
+     * Create a test case folder
+     *
+     * @param testCaseFolder The test case your want to create
+     *
+     * @return The created test case folder
+     */
+    public TestCaseFolder create(TestCaseFolder testCaseFolder) throws IOException {
```

Comment:
A proper method for creating a folder nested into an other folder is missing

Reply:
It should be possible to fill the parentId directly on the model, but for some convenience I added a createSubfolder method that will do the trick.